crates/types/src/lib.rs

```diff
@@ -87,10 +161,10 @@ pub struct WasmFuncType {
 impl WasmFuncType {
     #[inline]
     pub fn new(params: Box<[WasmType]>, returns: Box<[WasmType]>) -> Self {
-        let externref_params_count = params.iter().filter(|p| **p == WasmType::ExternRef).count();
+        let externref_params_count = params.iter().filter(|p| match **p { WasmType::Ref(WasmRefType { heap_type: WasmHeapType::Extern, .. }) => true, _ => false }).count();
         let externref_returns_count = returns
             .iter()
-            .filter(|r| **r == WasmType::ExternRef)
+            .filter(|r| match **r { WasmType::Ref(WasmRefType { heap_type: WasmHeapType::Extern, .. }) => true, _ => false })
```

Comment:
Speaking of which do non-nullable extern refs exist?

Reply:
I don't think so, as `externref` and `funcref` are interpreted as nullable refs c.f. https://github.com/WebAssembly/function-references/blob/main/proposals/function-references/Overview.md#Reference-Types.